server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -139,83 +313,96 @@ class InteractiveSession(
       mergeHiveSiteAndHiveDeps(sparkMajorVersion)
     }
 
-    val userOpts: Map[String, Option[String]] = Map(
-      "spark.driver.cores" -> request.driverCores.map(_.toString),
-      SparkLauncher.DRIVER_MEMORY -> request.driverMemory.map(_.toString),
-      SparkLauncher.EXECUTOR_CORES -> request.executorCores.map(_.toString),
-      SparkLauncher.EXECUTOR_MEMORY -> request.executorMemory.map(_.toString),
-      "spark.executor.instances" -> request.numExecutors.map(_.toString)
-    )
+    builderProperties
+  }
+}
 
-    userOpts.foreach { case (key, opt) =>
-      opt.foreach { value => builderProperties.put(key, value) }
-    }
+class InteractiveSession(
```

Comment:
I agree but mind if I do it in another PR? I want to keep Batch and Interactive code similar/symmetric. Since this's already a big PR, I don't want to make it even bigger.